backend/FwHeadless/Program.cs

```diff
@@ -146,10 +148,27 @@ static async Task<Results<Ok<ProjectSyncStatus>, NotFound>> GetMergeStatus(
         if (await projectLookupService.ProjectExists(projectId)) return TypedResults.Ok(ProjectSyncStatus.NeverSynced);
         else return TypedResults.NotFound();
     }
-    var commitsOnServer = await lexBoxDb.Set<ServerCommit>().CountAsync(c => c.ProjectId == projectId);
+    var lexboxProject = await lexBoxDb.Projects.Include(p => p.FlexProjectMetadata).FirstOrDefaultAsync(p => p.Id == projectId);
+    if (lexboxProject is null)
+    {
+        // Can't sync if lexbox doesn't have this project
+        return TypedResults.NotFound();
+    }
+    var crdtCommitsOnServer = await lexBoxDb.Set<ServerCommit>().CountAsync(c => c.ProjectId == projectId);
     var lcmCrdtDbContext = services.GetRequiredService<LcmCrdtDbContext>();
-    var localCommits = await lcmCrdtDbContext.Set<Commit>().CountAsync();
-    return TypedResults.Ok(ProjectSyncStatus.ReadyToSync(commitsOnServer - localCommits));
+    var localCrdtCommits = await lcmCrdtDbContext.Set<Commit>().CountAsync();
+    var pendingCrdtCommits = crdtCommitsOnServer - localCrdtCommits;
+    var lastCrdtCommitDate = await lcmCrdtDbContext.Set<Commit>().MaxAsync(commit => commit.DateTime);
+
+    var lastHgCommitDate = lexboxProject.LastCommit ?? DateTimeOffset.UnixEpoch;
```

Comment:
use null here too if we don't know the last commit

Reply:
Done in commit e6609784.